DataFormats/Math/interface/deltaPhi.h

```diff
@@ -4,16 +4,23 @@
  *
  * Ported from original code in RecoJets 
  * by Fedor Ratnikov, FNAL
+ * stabilize range reduction
  */
 #include <cmath>
 
 namespace reco {
+  
+  // reduce to [-pi,pi]
+  template<typename T> inline
+  T  reduceRange(T x) {
+   constexpr T o2pi = 1./(2.*M_PI);
+   if (std::abs(x) <= T(M_PI)) return x;
+   T n = std::round(x*o2pi);
+   return x - n*T(2.*M_PI);
+  }
 
   inline double deltaPhi(double phi1, double phi2) { 
-    double result = phi1 - phi2;
-    while (result > M_PI) result -= 2*M_PI;
-    while (result <= -M_PI) result += 2*M_PI;
-    return result;
+    return reduceRange(phi1 - phi2);
```

Comment:
did the code get faster by elimination of while?
if not, maybe a simple single 2pi flip should be added first if above/below +/-pi (this will cover almost all use cases)
and the rest can use the "round"